draft-ietf-mls-protocol.md

```diff
@@ -2135,23 +2131,74 @@ welcome_key = HKDF-Expand(welcome_secret, "key", key_length)
   particular, the confirmed transcript hash for the new state is the
   `prior_confirmed_transcript_hash` in the GroupInfo object.
 
-* Process the `path` field in the GroupInfo to update the new group state:
+    * Update the leaf at index `index` with the private key corresponding to the
+      public key in the node.
 
-   * Update the ratchet tree by replacing nodes in the direct path of the sender
-     with the corresponding nodes in the path (see {{direct-paths}}).
+    * Identify the lowest common ancestor of the leaves at `index` and at
+      `GroupInfo.signer_index`.  Set the private key for this node to the
+      private key derived from the `path_secret` in the KeyPackage object.
 
-   * The `commit_secret` is the value `path_secret[n+1]` derived from the
-     `path_secret[n]` value associated to the root node.
+    * For each parent of the common ancestor, up to the root of the tree, derive
+      a new path secret and set the private key for the node to the private key
+      derived from the path secret.  The private key MUST be the private key
+      that correspondns to the public key in the node.
 
-* Use the `epoch_secret` from the KeyPackage object to generate the epoch secret and other derived secrets for the
-  current epoch.
+* Use the `epoch_secret` from the KeyPackage object to generate the epoch secret
+  and other derived secrets for the current epoch.
 
 * Set the confirmed transcript hash in the new state to the value of the
   `confirmed_transcript_hash` in the GroupInfo.
 
 * Verify the confirmation MAC in the GroupInfo using the derived confirmation
   key and the `confirmed_transcript_hash` from the GroupInfo.
 
+# Extensibility
```

Comment:
This means we need some more text in the IANA considerations about setting up the space.